apps/next-app/src/components/painter/overlay/undo-button.tsx

```diff
@@ -0,0 +1,17 @@
+"use client";
+
+import { useActionHistory } from "../action-history";
+
+export function UndoButton(): JSX.Element {
```

Comment:
Can we make this one component with props?
Ex:

`export function HistoryButton(props: PropsWithChildren<{redo: boolean}>): JSX.Element {...}`

and then we can use `props.children` below to utilize server-side rendering.

Ex:

```
return (
  <button
    ...
  >
    {props.children}
  </button>
);
```

Reply:
Then, when instantiating the button you can pass the text in as a child, this will enable server-rendering the text.

Ex:

```
<HistoryButton>
  Undo
</HistoryButton>
```